tests/epyccel/test_epyccel_augassign.py

```diff
@@ -0,0 +1,170 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+
+import numpy as np
+import modules.augassign as mod
+
+from pyccel.epyccel import epyccel
+
+
+# += tests
+
+def test_augassign_add_1d(language):
+    f_int     = mod.augassign_add_1d_int
+    f_float   = mod.augassign_add_1d_float
+    f_complex = mod.augassign_add_1d_complex
+    f_int_epyc     = epyccel(f_int, language = language)
+    f_float_epyc   = epyccel(f_float, language = language)
+    f_complex_epyc = epyccel(f_complex, language = language)
+
+    x1_int     = np.ones(5, dtype=int)
+    x1_float   = np.ones(5, dtype=float)
+    x1_complex = np.ones(5, dtype=complex)
+    x2_int     = np.ones(5, dtype=int)
+    x2_float   = np.ones(5, dtype=float)
+    x2_complex = np.ones(5, dtype=complex)
+
+    assert f_int(x1_int) == f_int_epyc(x2_int) == 42
+    assert f_float(x1_float) == f_float_epyc(x2_float) == 4.2
+    assert f_complex(x1_complex) == f_complex_epyc(x2_complex) == (4.0 + 2.0j)
```

Comment:
It is not necessary to specify the  value that you think will be returned the pure python execution of the function is there to obtain this value.

By the way you need:
```python
    x1_int     = np.zeros(5, dtype=int)
    x1_float   = np.zeros(5, dtype=float)
    x1_complex = np.zeros(5, dtype=complex)
    x2_int     = np.zeros(5, dtype=int)
    x2_float   = np.zeros(5, dtype=float)
    x2_complex = np.zeros(5, dtype=complex)
```

to obtain the result you expect.

It would also be good to check that the pointer is working correctly and that the argument has indeed changed

Reply:
> It would also be good to check that the pointer is working correctly and that the argument has indeed changed
What pointer? I don't see what you mean.

Reply:
```
b = a
```
is not a copy. `b` points at `a`. So when `b` is modified, `a` should also be modified. It would be good to check that:
```python
np.all_equal(x1_int, x2_int)
```